modules/configuration/src/main/java/org/apache/ignite/configuration/internal/DynamicConfiguration.java

```diff
@@ -108,43 +85,50 @@ protected DynamicConfiguration(
     ) {
         members.put(member.key(), member);
 
-        configurator.addValidations((Class<? extends ConfigurationTree<?, ?>>) getClass(), member.key(), validators);
+//        configurator.addValidations((Class<? extends ConfigurationTree<?, ?>>) getClass(), member.key(), validators);
     }
 
     /** {@inheritDoc} */
-    @Override public void change(CHANGE change) throws ConfigurationValidationException {
-        configurator.set(BaseSelectors.find(qualifiedName), change);
+    @Override public Future<Void> change(Consumer<CHANGE> change) throws ConfigurationValidationException {
+        Objects.requireNonNull(change, "Configuration consumer cannot be null.");
+
+        InnerNode rootNodeChange = ((RootKeyImpl)rootKey).createRootNode();
+
+        if (keys.size() == 1)
+            change.accept((CHANGE)rootNodeChange);
+        else {
+            // TODO Not optimal, can be improved. Do it when tests are ready.
+            fillFromPrefixMap(rootNodeChange, toPrefixMap(Collections.singletonMap(join(keys), null)));
```

Comment:
this code needs to be refactored for more clarity

Reply:
Done, I hope it's better now